src/uu/split/src/split.rs

```diff
@@ -52,14 +52,86 @@ const AFTER_HELP: &str = help_section!("after help", "split.md");
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let args = args.collect_lossy();
+
+    let (args, obs_lines) = handle_obsolete(&args[..]);
+
     let matches = uu_app().try_get_matches_from(args)?;
-    match Settings::from(&matches) {
+
+    match Settings::from(&matches, &obs_lines) {
         Ok(settings) => split(&settings),
         Err(e) if e.requires_usage() => Err(UUsageError::new(1, format!("{e}"))),
         Err(e) => Err(USimpleError::new(1, format!("{e}"))),
     }
 }
 
+/// Extract obsolete shorthand (if any) for specifying lines in following scenarios (and similar)
+/// `split -22 file` would mean `split -l 22 file`
+/// `split -2de file` would mean `split -l 2 -d -e file`
+/// `split -x300e file` would mean `split -x -l 300 -e file`
+/// `split -x300e -22 file` would mean `split -x -e -l 22 file` (last obsolete lines option wins)
+/// following GNU `split` behavior
+fn handle_obsolete(args: &[String]) -> (Vec<String>, Option<String>) {
+    let mut obs_lines = None;
+    let filtered_args = args
+        .iter()
+        .filter_map(|slice| {
```

Comment:
This method is still no bulletproof unfortunately. The problem is that something that looks like a short option chunk could be an argument to a long flag. For example:
```
split --additional-suffix -x400a4
```
With this PR, this is the output I get:
```
error: a value is required for '--additional-suffix <SUFFIX>' but none was supplied

For more information, try '--help'.
```
That's quite confusing...

I think that case is more important to get exactly right than the shortcut.

I _think_ the way that GNU `split` does it, is by storing a number and defining short flags for each digit and when that digit in encountered multiply the number by 10 and adding the digit. You can see this too because `split -1 -2` behaves like `split -12`. I personally think that behaviour is more confusing than helpful, maybe it's the way to go?

You've done a great job with all the case you have taken into account here already, but this method might be a dead end if we want full compatibility. I'd be happy to be proven wrong though!

Reply:
Thanks for catching that! I will go deeper into the rabbit hole :)
On the `split -1 -2 file` example though - the GNU split would treat it as `split -2 file` , i.e. it will not combine `-1` with `-2` for a total `-12`, so at least that part works now :)

Reply:
It won't? Oh interesting! It's even more clever than I thought then.

Reply:
updated the code to handle this edge case correctly (in a generic way, should work for all options where it applies) + tests 
```
split --additional-suffix -x400a4
```